### DISCUSSION TITLE
is it possible to inject my own dependency on handle_spin function

### DISCUSSION THREAD

**Discussion by goldenratio** [spinframework/spin#2386] [General]
```
#[http_component]
fn handle_spin(req: Request, game_manager: MyOwnDependency) -> anyhow::Result<impl IntoResponse> {
}
```
wondering, if it is possible to DI my own dependencies in handle_spin function (example: game_manager: MyOwnDependency)

**Comment by seungjin** [comment]
Not sure how/why you want to inject into http_component.
One of my use cases was injecting my thing into router. 
```
let mut router = Router::new();
router.any_async("/api/v1/streaming/health", streaming::health::request);
```
Above is plain router example. But I wanted 

```
router.any_async("/api/v1/streaming/health", streaming::health::request, myOwnExtraStuff);
```
and my request has 
```
pub async fn request(req: Request, params: Params, myOwnExtraStuff: Something) -> Result<impl IntoResponse> {
```

So I can ship some business logics or session releated stuffs from very early stage of route.

**Comment by seungjin** [comment]
Also manipulation headers can be consider for some cases. 
https://discord.com/channels/926888690310053918/1185439992626032692

**Comment by lann** [comment]
I think we'd much rather support existing frameworks for this kind of advanced functionality. For example, I have a demo of [axum](https://crates.io/crates/axum) working with Spin that would give you this feature and many more: https://github.com/lann/wasi-hyperium/tree/main/examples/axum-server

**Reply by goldenratio** [reply]
awesome thank you! This is what I was looking for.

**Reply by goldenratio** [reply]
I don't see any spin reference there.. no spin.toml, etc..

**Reply by lann** [reply]
It isn't spin-specific; the same code will work in e.g. `wasmtime serve`. You should be able to use it with any example Rust spin.toml like https://github.com/fermyon/spin/blob/main/examples/http-rust/spin.toml (replacing `http_rust.wasm` with `axum_server.wasm` and `/hello` with `/...`).

Please keep in mind that this is only a demo at this point. I don't _know_ of anything that is broken with it but it isn't tested enough for me to have much confidence in that.
